build/Linux+BSD/portable/copy-libs

```diff
@@ -102,6 +107,123 @@ getCrossPlatformDependencies() {
+  copyLib libgstinterfaces-0.10.so.0
+  copyLib libgstpbutils-0.10.so.0
+  copyLib libgstreamer-0.10.so.0
+  copyLib libgstvideo-0.10.so.0
+  copyLib libharfbuzz.so.0
+  copyLib libICE.so.6
+  copyLib liblcms2.so.2
+  copyLib libogg.so.0
+  copyLib libSM.so.6
+  copyLib libsndfile.so.1
+  copyLib libsqlite3.so.0
+  copyLib libtiff.so.5
+  copyLib libvorbis.so.0
+  copyLib libvorbisfile.so.3
+  copyLib libwebp.so.5
+  copyLib libX11-xcb.so.1
+  copyLib libX11.so.6
+  copyLib libxcb-icccm.so.4
+  copyLib libxcb-image.so.0
+  copyLib libxcb-keysyms.so.1
+  copyLib libxcb-randr.so.0
+  copyLib libxcb-render-util.so.0
+  copyLib libxcb-render.so.0
+  copyLib libxcb-shape.so.0
+  copyLib libxcb-shm.so.0
+  copyLib libxcb-sync.so.1
+  copyLib libxcb-xfixes.so.0
+  copyLib libxcb-xkb.so.1
+  copyLib libxcb.so.1
+  copyLib libXi.so.6
+  copyLib libxkbcommon-x11.so.0
+  copyLib libxkbcommon.so.0
+  copyLib libxml2.so.2
+  copyLib libxslt.so.1
+
+  #2nd round of dependencies
+  copyLib libavahi-client.so.3
+  copyLib libavahi-common.so.3
+  copyLib libdrm.so.2
+  copyLib libexpat.so && mv "$dest_dir/libexpat.so" "$dest_dir/libexpat.so.1" #renamed to .1 cause didn't have
+  copyLib libffi.so.6
+  copyLib libFLAC.so.8
+  copyLib libgbm.so.1
+  copyLib libglapi.so.0
+  copyLib libgmodule-2.0.so.0
+  copyLib libgnutls-deb0.so.28
+  copyLib libgraphite2.so.3
+  copyLib libgssapi_krb5.so.2
+  copyLib libgthread-2.0.so.0
+  copyLib libjbig.so.0
+  copyLib libjson-c.so && mv "$dest_dir/libjson-c.so" "$dest_dir/libjson-c.so.2" #renamed to .2 cause didn't have
+  copyLib liborc-0.4.so.0
+  copyLib libvorbisenc.so.2
+  copyLib libwayland-client.so.0
+  copyLib libwayland-server.so.0
+  copyLib libXau.so.6
+  copyLib libxcb-dri2.so.0
+  copyLib libxcb-util.so.0
+  copyLib libXdmcp.so.6
+  copyLib libXext.so.6
+
+  #3rd round of dependencies
+  copyLib libasyncns.so.0
+  copyLib libgmp.so.10
+  copyLib libhogweed.so.2
+  copyLib libjack.so.0
+  copyLib libjpeg.so.62
+  copyLib libk5crypto.so.3
+  copyLib libkeyutils.so && mv "$dest_dir/libkeyutils.so" "$dest_dir/libkeyutils.so.1" #renamed to .1 cause didn't have
+  copyLib libkrb5.so.3
+  copyLib libkrb5support.so.0
+  copyLib libmng.so.1
+  copyLib libnettle.so.4
+  copyLib libp11-kit.so.0
+  copyLib libpng12.so.0
+
+  copyLib libtasn1.so.6
+  copyLib libwrap.so.0
+  copyLib libXtst.so.6
+  dest_dir="$qt_dest/lib"
+  copyLib libQt5Concurrent.so.5
+  copyLib libQt5DBus.so.5
+  copyLib libQt5Designer.so.5
+  copyLib libQt5QuickWidgets.so.5
+  copyLib libQt5Test.so.5
+
+  #jessie doesn't seem to have the following, although in sid this is provided by libqt5gui5
+  #copyLib libQt5XcbQpa.so.5
+  #copyQt plugins/xcbglintegrations/libqxcb-glx-integration.so # Needed by plugins which open a window
+
+  #needed by Arch Linux
+  dest_dir="$lib_dest"
+  copyLib libpcre.so.3
+
+  #needed if don't have pulseaudio installed
+  copyLib libpulse.so.0
+  copyLib libpulsecommon-5.0.so
+  copyLib libasound_module_pcm_pulse.so
```

Comment:
You can try it and see...
It generally doesn't matter where libraries go as long as the location is in the cache, LD_LIBRARY_PATH or LD_PRELOAD. Alternatively, they could also go in the same folder as the executable. However, there is one exception that I know of: if the library location has been hard-coded into the executable using the RPATH variable. Qt libraries all have RPATH defined, hence the Qt directory structure has to be preserved in the "lib" folder inside the AppImage. You can use the program `chrpath` to see if a given library or binary has RPATH set and to print it if they do. Of course, you don't know which library has it set so you'd need to do it for all of them. Run this in the "lib" folder inside the AppDir.
```bash
# Show all RPATH for all libraries that have it set
find . -exec bash -c "chrpath {} &>/dev/null" \; -exec chrpath {} \;
# Find the one pointing to libasound_module_pcm_pulse.so
find . -exec bash -c "chrpath {} &>/dev/null" \; -exec chrpath {} \; | grep libasound_module_pcm_pulse.so
```